src/index/IndexImpl.Text.cpp

```diff
@@ -99,31 +99,21 @@ void IndexImpl::logEntityNotFound(const string& word,
 }
 
 // _____________________________________________________________________________
-void IndexImpl::buildTextIndexFile(TextIndexConfig config) {
-  config.checkValid();
-  bool addWordsFromFiles =
-      config.docsFile_.has_value() &&
-      (config.wordsFile_.has_value() || config.useDocsFileForVocabulary_);
-  if (config.addOnlyEntitiesFromWordsFile_ &&
-      !config.useDocsFileForVocabulary_) {
-    throw std::runtime_error(
-        "The option to add entities from wordsfile can only be used together "
-        "with the option to build the text index from docsfile since "
-        "otherwise the wordsfile would be used anyway.");
-  }
-  string wordsFile = config.wordsFile_.value_or("");
-  string docsFile = config.docsFile_.value_or("");
+void IndexImpl::buildTextIndexFile(TextIndexConfig&& textIndexConfig) {
+  const auto config = TextIndexConfig(std::move(textIndexConfig));
```

Comment:
That is not really necessary.
The parameter can just be `TextIndexConfig config` (no `&&` , but still consistently `std::move()`).